kong/clustering/compat/checkers.lua

```diff
@@ -40,37 +40,43 @@ local compatible_checkers = {
 
         if plugin.name == 'ai-proxy' then
           local config = plugin.config
-          if config.model.provider == "gemini" then
+          if config.model.provider == "gemini" or config.model.provider == "bedrock" then
+            log_warn_message('configures ' .. plugin.name .. ' plugin with' ..
+            ' "openai preserve mode", because ' .. config.model.provider .. ' provider ' ..
+            ' is not supported in this release',
+            dp_version, log_suffix)
+
             config.model.provider = "openai"
             config.route_type = "preserve"
-            log_warn_message('configures ' .. plugin.name .. ' plugin with' ..
-                              ' "openai preserve mode", because gemini' ..
-                              ' provider is not supported in this release',
-                              dp_version, log_suffix)
+
             has_update = true
           end
         end
 
         if plugin.name == 'ai-request-transformer' then
           local config = plugin.config
-          if config.llm.model.provider == "gemini" then
-            config.llm.model.provider = "openai"
+          if config.llm.model.provider == "gemini" or config.llm.model.provider == "bedrock" then
```

Comment:
You also seem to do quite a bit of comparison for model/providers. Consider doing something like

``` lua
local GEMINI_PROVIDER_NAME = "gemini"

is_gemini = function()
  return config.llm.model.provider == GEMINI_PROVIDER_NAME
end

-- allows you to do compounds as well

is_gemini_or_bedrock() = funtion()
  return is_gemini() or is_bedrock()
end

```

Reply:
But will this JIT really badly, and have bad performance? Is there anywhere else I can copy from for this?

Reply:
I have an idea

Reply:
What about this?

https://github.com/Kong/kong/pull/13354/files#diff-5df18abd0db44afd52ca7f705ec37e8ebbec7640e065941fef133e412eb885b4R23

https://github.com/Kong/kong/pull/13354/files#diff-5df18abd0db44afd52ca7f705ec37e8ebbec7640e065941fef133e412eb885b4R59

Keeps strings together, and also jits all strings at startup (when clustering package launches).

Reply:
I've done what Josh suggested, see https://github.com/Kong/kong/pull/13354/files#diff-5df18abd0db44afd52ca7f705ec37e8ebbec7640e065941fef133e412eb885b4R23